x/evm/keeper/state_transition.go

```diff
@@ -166,17 +167,11 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 
 	res.Hash = txHash.Hex()
 	logs := k.GetTxLogs(txHash)
+	k.CommitCachedContexts()
```

Comment:
so we are not checking if the res failed? can you explain again why is this the case and add a comment on the code?

Reply:
Because the evm will call `RevertToSnapshot` when failure, so basically here we assume evm internal will handle the state revert properly, so for example when failure the `logs` should always be empty I suppose.

Reply:
> the evm will call RevertToSnapshot when failure, so basically here we assume evm internal already handled the state revert properly

this should be written as a comment on the code

Reply:
done